custom_model_runner/datarobot_drum/drum/language_predictors/base_language_predictor.py

```diff
@@ -188,25 +202,39 @@ def monitor(self, kwargs, predictions, predict_time_ms):
                 num_predictions=len(predictions), execution_time_ms=predict_time_ms
             )
 
+            if self._training_data_available:
+                df = StructuredInputReadUtils.read_structured_input_data_as_df(
+                    kwargs.get(StructuredDtoKeys.BINARY_DATA),
+                    kwargs.get(StructuredDtoKeys.MIMETYPE),
+                )
+            else:
+                # Feature drift and predictions data collections is disabled, no point
+                # reporting it.
+                df = None
             # TODO: Need to convert predictions to a proper format
             # TODO: or add report_predictions_data that can handle a df directly..
             # TODO: need to handle associds correctly
 
-            # mlops.report_predictions_data expect the prediction data in the following format:
-            # Regression: [10, 12, 13]
-            # Classification: [[0.5, 0.5], [0.7, 03]]
-            # In case of classification, class names are also required
-            class_names = None
-            if len(predictions.columns) == 1:
-                mlops_predictions = predictions[predictions.columns[0]].tolist()
+            if self._training_data_available:
+                # mlops.report_predictions_data expect the prediction data in the following format:
+                # Regression: [10, 12, 13]
+                # Classification: [[0.5, 0.5], [0.7, 03]]
+                # In case of classification, class names are also required
+                class_names = None
+                if len(predictions.columns) == 1:
+                    mlops_predictions = predictions[predictions.columns[0]].tolist()
+                else:
+                    mlops_predictions = predictions.values.tolist()
+                    class_names = list(predictions.columns)
             else:
-                mlops_predictions = predictions.values.tolist()
-                class_names = list(predictions.columns)
+                # Target drift is disabled, no point reporting predictions for monitoring
+                mlops_predictions = None
+                class_names = None
+
+            if mlops_predictions is None and df is None:
```

Comment:
Could it be that `mlops_predictions` is an empty list? If yes, I'd change it to:
```suggestion
            if not mlops_predictions and df is None:
```

Reply:
`mlops_predictions` won't be empty - If there is a prediction request - there should be some prediction.